### PR TITLE
Ensure guiders are copied to new location on edit

### PR DESCRIPTION
**Pull request by dwhenry**
Previously editing a location would result in all
the guiders associated with it being left associated
with the old version of the location.

### CODE REVIEW

File: spec/services/update_location_spec.rb [hunk 1]
```diff
@@ -36,6 +36,23 @@
     end
 
     context 'when an existing location would be changed by the update' do
+      context 'and the location has guiders assigned to it' do
+        let(:guider) { create(:guider, location: location) }
+
+        before do
+          guider
```
Comment (benlovell) [inline spec/services/update_location_spec.rb]: May as well make the `let` a `let!` and drop this.

File: app/services/create_or_update_location.rb [hunk 2]
```diff
@@ -18,11 +18,7 @@ def build(params)
   end
 
   def update(params)
-    address = params.delete('address')
-    if address
-      address_params = build_address(address)
-      params.merge!(address_params)
-    end
+    params.merge!(build_address(params.delete('address')))
```
Comment (benlovell) [inline app/services/create_or_update_location.rb]: There is a bunch of refactoring that doesn't really belong in this commit with the message it has. Can you pluck it out so that the guider stuff is in a single commit?

File: app/services/create_or_update_location.rb [hunk 3]
```diff
@@ -58,8 +55,12 @@ def build_new_version(params)
   end
 
   def create_new_version(params)
+    guiders = location.guiders
     build_new_version(params)
     @location.save!
+    guiders.each do |guider|
```
Comment (benlovell) [inline app/services/create_or_update_location.rb]: I wonder if it just makes sense to just `update_all(location_id: X)` here instead of copying? Is there a case when prior versions are restored?

**Comment by dwhenry** [comment]
@benlovell updated.. :)
